python/ray/data/tests/test_formats.py

```diff
@@ -162,6 +166,33 @@ def test_write_datasource(ray_start_regular_shared):
     assert ray.get(output.data_sink.get_rows_written.remote()) == 10
 
 
+def test_open_file_with_retry(ray_start_regular_shared):
```

Comment:
it's kind of weird to put this test in `test_formats.py`

Reply:
good catch, moved to `test_file_based_datasource.py`.